R/dtc_utils.R

```diff
@@ -0,0 +1,204 @@
+#' Assert date time character formats
+#'
+#' [assert_dtc_fmt()] takes a character vector of date/time formats and checks if
+#' the formats are supported, meaning it checks if they are one of the formats
+#' listed in column `fmt` of [dtc_formats], failing with an error otherwise.
+#'
+#' @param fmt A character vector.
+#'
+#' @examples
+#' sdtm.oak:::assert_dtc_fmt(c("ymd", "y m d", "dmy", "HM", "H:M:S", "y-m-d H:M:S"))
+#'
+#' # This example is guarded to avoid throwing errors
+#' if (FALSE) {
+#'   sdtm.oak:::assert_dtc_fmt("y years m months d days")
+#' }
+#'
+#' @keywords internal
+assert_dtc_fmt <- function(fmt) {
+  admiraldev::assert_character_vector(fmt)
```

Comment:
We do need `assert_character_vectora` as it should be covered by `rlang::arg_match`. Is there any reason for using it here?

Reply:
There is no need indeed, I will remove it! Thanks for spotting this.

Reply:
Resolved in https://github.com/pharmaverse/sdtm.oak/pull/21/commits/d6cd751dfd06d9747cfa528974c7ae449b381d01.